### DISCUSSION TITLE
Access to the default theme components

### DISCUSSION THREAD

**Discussion by erondpowell** [vuejs/vitepress#824] [Q&A]
Hello,

I'm going to modify/extend the default theme and I may want access to the original components.

I'm aware that in VuePress it was possible to bring in the original components to access/modify them in the working directory.
Is this possible/necessary for VitePress? What is the best practice for VitePress?

Thanks so much!

**Comment by kiaking** [comment]
Currently there're only few components exposed from `vitepress/theme`, but you could directly access the theme component at `vitepress/dist/client/theme-default` so that's one way of doing it.

But unexposed component means that those might change in the future in a way that breaks your extended component

It depends how deeply you would like to modify the original component. If you need to change them big, then I would recommend not to extending the components but just copy and paste the source from VitePress to your own components. That way, you can feel more safe when upgrading VitePress in the future.

If you're planning just a small change but some feature is missing, let's open a discussion for adding that specific feature 👍

**Reply by erondpowell** [reply]
okay, thanks for the answer @kiaking!
